R/dash.R

```diff
@@ -374,55 +344,24 @@ Dash <- R6::R6Class(
     # ------------------------------------------------------------------------
     # callback registration
     # ------------------------------------------------------------------------
-    callback = function(func = NULL, output = NULL) {
+    callback = function(output = NULL, inputs = list(), state = list(), user_function) {
+      private$layout_render()
```

Comment:
Why is this called?

Reply:
If you comment out that line, the app does not finish rendering. This line was originally present in Carson's code (see below); I deleted it after you and I spoke last week, but reinserted it after I noticed the page never finished loading.

https://github.com/plotly/dashR/blob/41e53e3cc5bdfcbfb6e94373ac28d4939b001457/R/dash.R#L384

Reply:
If we can't figure this out in this PR please make an issue for it. For a complicated app this would be a major drain, re-rendering the layout for every callback you attach. It'll only happen on app initialization, but if you have a slow layout and lots of callbacks it would be painful.

dash-py does renders the layout once in `layout.setter` - perhaps we can remove it here and add it to `layout_set`?

Reply:
As discussed earlier today, will attempt to relocate the `layout_render()` call to the `layout_set()` method.